### PR TITLE
feat(format): better checking of format options.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- check format by format:

1.   report errors when option that should not use, e.g.  `field_delimeter` for `Parquet`
2.  set format`s own default if meet "" for some options.

- real enable option nan_display
- almost remove the usage of FormatSetting, leave alone the usage in Variant, which will be remove after new expression is used  https://github.com/datafuselabs/databend/issues/8949
- some other refactor.

Closes https://github.com/datafuselabs/databend/issues/8949  https://github.com/datafuselabs/databend/issues/8935

### CODE REVIEW

File: src/common/io/src/format_settings.rs [hunk 1]
```diff
@@ -13,145 +13,17 @@
...
-            empty_as_default: true,
-            json_quote_denormals: false,
-            json_escape_forward_slashes: true,
-            ident_case_sensitive: false,
-            row_tag: vec![],
-        }
-    }
 }
 
 // only used for tests
```
Comment (ariesdevil) [inline src/common/io/src/format_settings.rs:22]: Still only used for tests?

Comment (youngsofun) [reply src/common/io/src/format_settings.rs:22]: used here https://github.com/datafuselabs/databend/issues/8949

File: src/query/formats/src/field_encoder/csv.rs [hunk 2]
```diff
@@ -39,7 +38,7 @@ impl FieldEncoderCSV {
                 true_bytes: TRUE_BYTES_LOWER.as_bytes().to_vec(),
                 false_bytes: FALSE_BYTES_LOWER.as_bytes().to_vec(),
                 null_bytes: NULL_BYTES_ESCAPE.as_bytes().to_vec(),
-                nan_bytes: NAN_BYTES_SNAKE.as_bytes().to_vec(),
+                nan_bytes: options.stage.nan_display.as_bytes().to_vec(),
```
Comment (ariesdevil) [inline src/query/formats/src/field_encoder/csv.rs:41]: Can we unify encode and only decode let the user choose?

File: src/query/formats/src/field_encoder/tsv.rs [hunk 3]
```diff
@@ -37,7 +36,7 @@ impl FieldEncoderTSV {
                 true_bytes: TRUE_BYTES_NUM.as_bytes().to_vec(),
                 false_bytes: FALSE_BYTES_NUM.as_bytes().to_vec(),
                 null_bytes: NULL_BYTES_ESCAPE.as_bytes().to_vec(),
-                nan_bytes: NAN_BYTES_LOWER.as_bytes().to_vec(),
+                nan_bytes: options.stage.nan_display.as_bytes().to_vec(),
```
Comment (ariesdevil) [inline src/query/formats/src/field_encoder/tsv.rs:39]: same question as above

File: src/query/formats/src/field_encoder/csv.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/formats/src/field_encoder/csv.rs:41]: [clickhouse TSV may require `nan`,  we have to make clickhouse driver work](https://github.com/datafuselabs/databend/discussions/8934#discussioncomment-4241662)

File: src/query/formats/src/field_encoder/tsv.rs (hunk 3 above)
Comment (youngsofun) [reply src/query/formats/src/field_encoder/tsv.rs:39]: https://github.com/datafuselabs/databend/discussions/8934#discussioncomment-4241662

File: src/query/formats/src/field_encoder/csv.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/formats/src/field_encoder/csv.rs:41]: this is another issue, and need to make sure it is safe before changing it.

**Review by ariesdevil — APPROVED** [review]

File: src/query/formats/src/field_encoder/csv.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/formats/src/field_encoder/csv.rs:41]: > this is another issue, and need to make sure it is safe before changing it.

I meant the opposite in the discuss:  
 user no need choose decode (load) if  we accept all common reprs.
they may want to choose encode(unload) to fullfil requirement of another system.

File: src/common/io/src/format_settings.rs [hunk 4]
```diff
@@ -13,145 +13,17 @@
...
-            inf_bytes: INF_BYTES_LOWER.as_bytes().to_vec(),
-            quote_char: b'\'',
-        }
-    }
-}
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct FormatSettings {
-    // both
     pub timezone: Tz,
```
Comment (TCeason) [inline src/common/io/src/format_settings.rs:19]: This strict only has one field timezone? Maybe we can extract it, and remove this strict?

Comment (TCeason) [reply src/common/io/src/format_settings.rs:19]: Will be refactor after new expression merged.

**Review by TCeason — APPROVED** [review]

File: src/common/io/src/format_settings.rs (hunk 4 above)
Comment (youngsofun) [reply src/common/io/src/format_settings.rs:19]: almost remove the usage of FormatSetting, leave alone the usage in Variant, which will be remove after new expression is used https://github.com/datafuselabs/databend/issues/8949

**Comment by BohuTANG** [comment]
Two failed tests need to be addressed:
Dev Linux / test_stateful_standalone_linux (pull_request) Failing after 8m
Dev Linux / test_stateful_cluster_linux (pull_request) Failing after 11m

**Review by sundy-li — APPROVED** [review]

**Pull request merged**